exporters/elasticsearch/src/es_log_recordable.cc

```diff
@@ -180,28 +70,21 @@ void ElasticSearchRecordable::SetTimestamp(
 #if __cplusplus >= 202002L
   const std::string dateStr = std::format("{:%FT%T%Ez}", timePoint);
 #else
-  const static int dateToSecondsSize = 19;
-  const static int millisecondsSize  = 8;
-  const static int timeZoneSize      = 1;
-  const static int dateSize          = dateToSecondsSize + millisecondsSize + timeZoneSize;
+  constexpr auto dateToSecondsSize = sizeof("YYYY-MM-DDTHH:MM:SS") - 1;
+  constexpr auto millisecondsSize  = sizeof(".123456") - 1;
+  constexpr auto timeZoneSize      = sizeof("Z") - 1;
+  constexpr auto dateSize          = dateToSecondsSize + millisecondsSize + timeZoneSize;
 
   std::time_t time = std::chrono::system_clock::to_time_t(timePoint);
   std::tm tm       = *std::gmtime(&time);
-
-  char bufferDate[dateSize];  // example: 2024-10-18T07:26:00.123456Z
-  std::strftime(bufferDate, sizeof(bufferDate), "%Y-%m-%dT%H:%M:%S", &tm);
   auto microseconds =
       std::chrono::duration_cast<std::chrono::microseconds>(timePoint.time_since_epoch()) %
       std::chrono::seconds(1);
 
-  char bufferMilliseconds[millisecondsSize];
-  std::snprintf(bufferMilliseconds, sizeof(bufferMilliseconds), ".%06ld",
+  char dateStr[dateSize + 1];  // example: 2024-10-18T07:26:00.123456Z
+  std::snprintf(dateStr, sizeof(dateStr), "%04d-%02d-%02dT%02d:%02d:%02d.%06ldZ", tm.tm_year + 1900,
+                tm.tm_mon + 1, tm.tm_mday, tm.tm_hour, tm.tm_min, tm.tm_sec,
```

Comment:
We can format the date in one go; there's no need to do unnecessary concatenations.

Also, because `nlohmann::json` handles `const char*`, there is no need to construct an `std::string`.